Bottom: 81bef6e680e2bed681dddb31c5b33cbfdfea202c
Top:    3b373b3dd3b9570d0b6d0a70b6c2026adfbb98d2
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-02-04 11:59:31 +0000

Attempt to fix broken polygon contour generation


---

diff --git a/src/polygon.h b/src/polygon.h
index 3737f35..e46f01e 100644
--- a/src/polygon.h
+++ b/src/polygon.h
@@ -34,7 +34,7 @@
 
 /* Implementation constants */
 
-#define POLY_CIRC_SEGS 40
+#define POLY_CIRC_SEGS 8
 #define POLY_CIRC_SEGS_F ((float)POLY_CIRC_SEGS)
 
 /* adjustment to make the segments outline the circle rather than connect
diff --git a/src/polygon1.c b/src/polygon1.c
index 352f8f6..87114de 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1686,7 +1686,7 @@ static void
 Collect (jmp_buf * e, PLINE * a, POLYAREA ** contours, PLINE ** holes,
 	 S_Rule s_rule, J_Rule j_rule)
 {
-  VNODE *cur, *other;
+  VNODE *cur;
   DIRECTION dir;
 
   cur = &a->head;
@@ -1694,9 +1694,6 @@ Collect (jmp_buf * e, PLINE * a, POLYAREA ** contours, PLINE ** holes,
     {
       if (s_rule (cur, &dir) && cur->Flags.mark == 0)
 	Collect1 (e, cur, dir, contours, holes, j_rule);
-      other = cur;
-      if ((other->cvc_prev && jump (&other, &dir, j_rule)))
-	Collect1 (e, other, dir, contours, holes, j_rule);
     }
   while ((cur = cur->next) != &a->head);
 }				/* Collect */
